runtime/doc/repeat.txt

```diff
@@ -172,42 +172,34 @@ Using Vim scripts					*using-scripts*
 For writing a Vim script, see chapter 41 of the user manual |usr_41.txt|.
 
 					*:so* *:source* *load-vim-script*
-:so[urce] {file}	Runs vim or lua {file}
```

Comment:
this wording is mushy. Let's be precise where we can, without being pedantic.

Reply:
Fyi I changed some of this in #14809, though I'm not sure which is preferred.

Reply:
the wording in this PR is less redundant and clearer. For example, #14809 still contains this extra passage:

```
			Note: Only files or buffers with names ending in
			`.lua` are sourced as Lua.  Anything else is assumed
			to be Vim script.
```

That's very verbose. This sentence is clear enough for users:

    Runs |Ex| commands or Lua code (".lua" files)

the `(".lua" files)` makes it obvious what's expected.

Reply:
I agree, this is definitely much more concise.

Reply:
There is no reason to mention that {file} is optional, unless it has a purpose. Normally optional arguments are specified by brackets in the documentation, like `[file]`. example:

    :ru[ntime][!] [where] {file} ..

Reply:
Eh, I think it's worth mentoning. Since #11444, `:source` will use the content of the current buffer if `{file}` is omitted. If the buffer name ends in ".lua" it'll be sourced as Lua instead.

One thing that I guess I should also be mentioned is that stuff like `s:` variables doesn't work in that case?

Reply:
oh wow, I forgot `:source` without args works on the current/unsaved buffer! That is awesome, and we definitely need to document it.